src/main/java/org/jabref/gui/fieldeditors/IdentifierEditor.java

```diff
@@ -85,6 +90,11 @@ private void lookupIdentifier() {
 
     @FXML
     private void openExternalLink() {
-        viewModel.openExternalLink();
+        if (StandardField.DOI.equals(field) && preferences.getDOIPreferences().isUseCustom()) {
+            String baseURL = preferences.getDOIPreferences().getDefaultBaseURI();
+            viewModel.openDOIWithCustomBase(baseURL);
+        } else {
+            viewModel.openExternalLink();
+        }
```

Comment:
Can you move this distinction to the viewModel?
(you probably have to inject the PreferencesService by constructor then too)